### PR TITLE
EPMRPP-81367 || Create plugin template

### PR DESCRIPTION
**Pull request by Bam6ycha**

### CODE REVIEW

File: plugin/ui/README.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
+# Billing plugin for Epam Report Portal
```
Comment (AmsterGet) [inline plugin/ui/README.md]: Plugin template

File: plugin/ui/webpack.config.js [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+      constants: path.resolve(__dirname, 'src/constants'),
+      icons: path.resolve(__dirname, 'src/icons'),
+      hooks: path.resolve(__dirname, 'src/hooks'),
+      utils: path.resolve(__dirname, 'src/utils'),
+      analyticsEvents: path.resolve(__dirname, 'src/analyticsEvents'),
+    },
+  },
+  externals: ['redux'],
+  plugins: [
+    new ForkTsCheckerWebpackPlugin({ typescript: { memoryLimit: 4096 } }),
```
Comment (AmsterGet) [inline plugin/ui/webpack.config.js]: Can we decrease memory limit for this process? 4gb looks too big

File: plugin/ui/webpack.config.js [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+          singleton: true,
+          requiredVersion: pjson.dependencies['classnames'],
+        },
+      },
+      exposes: {
+        './moduleName': './src/components/moduleName',
+      },
+    }),
+    new CopyPlugin({
+      patterns: [{ from: path.resolve(__dirname, './src/metadata.json') }],
```
Comment (AmsterGet) [inline plugin/ui/webpack.config.js]: Also please add a default icon for plugin.

File: plugin/ui/src/metadata.json [hunk 4]
```diff
@@ -0,0 +1,11 @@
+{
+  "scope": "plugin_name",
+  "extensions": [
+    {
+      "name": "plugin",
+      "type": "uiExtension:settingsTab",
+      "title": "Plugin",
+      "moduleName": "./moduleName"
```
Comment (AmsterGet) [inline plugin/ui/src/metadata.json:8]: Please add this module as an example with necessary exports and empty content.

File: plugin/ui/webpack.config.js [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+    new ModuleFederationPlugin({
+      name: 'plugin_name',
+      filename: `remoteEntity.js`,
+      shared: {
+        react: {
+          import: 'react',
+          shareKey: 'react',
+          shareScope: 'default',
+          singleton: true,
+          requiredVersion: pjson.dependencies['react'],
```
Comment (AmsterGet) [inline plugin/ui/webpack.config.js:80]: As I see it is not the best solution here, refer this [issue](https://github.com/webpack/webpack/issues/15971#issuecomment-1172163299).
But I guess you've fixed the warnings in the browser console related to the deps sharing. Am I right?

**Review by AmsterGet — CHANGES_REQUESTED** [review]

File: plugin/ui/webpack.config.js (hunk 5 above)
Comment (Bam6ycha) [reply plugin/ui/webpack.config.js:80]: Yes, you are right!

Comment (AmsterGet) [reply plugin/ui/webpack.config.js:80]: Ok, let's bypass this warnings.

**Review by AmsterGet — APPROVED** [review]

**Pull request merged**